File: C++/list_examples.cpp
#include <iostream>
#include <vector>

int main(int argc, const char * argv[]) {

  // Create a list
  std::vector<int> l = {1, 2, 3};

  // Add an item
  l.push_back(4);

  // Get length of the list
  printf("%lu\n", l.size());

  // Return the second item
  printf("%d\n", l[1]);

  // Remove the first item
  l.erase(l.begin());
  
  // Clear the list
  l.clear();

  return 0;
}

File: ObjectiveC/ListExamples.m
#import <Foundation/Foundation.h>

int main(const int argc, char **argv) {

  // Create a list
  NSMutableArray* l = [@[@1, @2, @3] mutableCopy];

  // Add an item
  [l addObject:@4];

  // Get length of the list
  NSLog(@"%lu", l.count);

  // Return the second item
  NSLog(@"%@", l[1]);

  // Remove the first item
  [l removeObjectAtIndex:0];

  // Clear the list
  [l removeAllObjects];

  return 0;
}
